mlflow/pytorch/__init__.py

```diff
@@ -769,9 +771,15 @@ class _PyTorchWrapper:
     def __init__(self, pytorch_model):
         self.pytorch_model = pytorch_model
 
-    def predict(self, data, device="cpu"):
+    def predict(self, data, device=_DEFAULT_PYTORCH_PREDICTION_DEVICE):
         import torch
-
+        if not MLFLOW_PYTORCH_DISABLE_AUTOMATIC_GPU_USAGE.get() and torch.cuda.is_available():
+            # if CUDA is available, we use the default CUDA device.
+            # To force inference to the CPU when the GPU is available, please set
+            # MLFLOW_PYTORCH_DISABLE_AUTOMATIC_GPU_USAGE to false
+            # If a specific non-default device is passed in, we continue to respect that.
+            if device == _DEFAULT_PYTORCH_PREDICTION_DEVICE:
```

Comment:
If I passed in device="cpu" and I had a cuda enabled bit of hardware on my machine, this wouldn't respect my declaration, would it?

Reply:
That's right - you'd have to use the env variable to disable the GPU placement.

Reply:
I'm not aware of a way to distinguish though - do you know how we might do that?

Reply:
could we have the `_DEFAULT_PYTORCH_PREDICTION_DEVICE` be a placeholder that gets overridden to 'cpu' if the default isn't overridden or the flag is `True` or cuda isn't available?

Reply:
That's a great point - we could do that

Reply:
I would rather we define an environment variable called `MLFLOW_DEFAULT_DEVICE` with default value `None` and then set:

```
_DEFAULT_PYTORCH_PREDICTION_DEVICE = MLFLOW_DEFAULT_DEVICE.get() or (
    "gpu" if torch.cuda.is_available() else "cpu"
)
```

Reply:
Updated to something along these lines - please lmk what you guys think